src/main/java/echoserver/Listenable.java

```diff
@@ -0,0 +1,8 @@
+package echoserver;
+
+import java.io.IOException;
+
+public interface Listenable {
+    Connection open() throws IOException;
+    void close() throws IOException;
```

Comment:
You might have a look at `AutoClosable` and try-with-resources

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html

You can make your  interface implement `AutoClosable` then you don't need to include `close` here and you can do things like this

```java
try (Listenable listenable = new Listener(serverSocket);) {
    // do stuff with listenable
}

// listenable is closed automatically once we get here
```

Reply:
Cool. Will do.